### PR TITLE
Zookeeper put api

### PR DESCRIPTION
**Pull request by kishoreg**
## Description
Adding an api to update ZK Path. Will be used by the UI
## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)
## Release Notes
If you have tagged this as either backward-incompat or release-notes,
you MUST add text here that you would like to see appear in release notes of the
next release.

If you have a series of commits adding or enabling a feature, then
add this section only in final commit that marks the feature completed.
Refer to earlier release notes to see examples of text

## Documentation
If you have introduced a new feature or configuration, please add it to the documentation as well.
See https://docs.pinot.apache.org/developers/developers-and-contributors/update-document

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java [hunk 1]
```diff
@@ -75,6 +80,63 @@ public String getData(
     return null;
   }
 
+  @DELETE
+  @Path("/zk/delete")
+  @Produces(MediaType.TEXT_PLAIN)
+  @ApiOperation(value = "Get content of the znode")
+  @ApiResponses(value = { //
```
Comment (mayankshriv) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java:87]: Are the trailing `//` needed for formatting, or some other purpose?

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java [hunk 2]
```diff
@@ -75,6 +80,63 @@ public String getData(
     return null;
   }
 
+  @DELETE
+  @Path("/zk/delete")
+  @Produces(MediaType.TEXT_PLAIN)
+  @ApiOperation(value = "Get content of the znode")
```
Comment (mayankshriv) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java:86]: `value` should be `delete` and not `Get`?

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java [hunk 3]
```diff
@@ -75,6 +80,63 @@ public String getData(
     return null;
   }
 
+  @DELETE
+  @Path("/zk/delete")
```
Comment (mayankshriv) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java:84]: The `delete` api seems a bit dangerous. Should we protect it to only allow deleting certain paths? Seems this api will allow deleting the entire cluster?

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java [hunk 4]
```diff
@@ -75,6 +80,63 @@ public String getData(
...
+      @ApiParam(value = "Content", required = true) @QueryParam("data") @DefaultValue("") String content,
+      @ApiParam(value = "expectedVersion", required = true, defaultValue = "-1") @QueryParam("expectedVersion") @DefaultValue("-1") String expectedVersion,
+      @ApiParam(value = "accessOption", required = true, defaultValue = "1") @QueryParam("accessOption") @DefaultValue("1") String accessOption) {
+    path = validateAndNormalizeZKPath(path);
+    ZNRecord record = null;
+    if (content != null) {
+      record = (ZNRecord) _znRecordSerializer.deserialize(content.getBytes(Charsets.UTF_8));
+    }
+    try {
+      boolean result = pinotHelixResourceManager
```
Comment (mayankshriv) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java:126]: Will this allow any arbitrary path to be set with any arbitrary value (as-in possibly corrupt the cluster state)?

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java (hunk 2 above)
Comment (kishoreg) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java]: fixed

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java (hunk 1 above)
Comment (kishoreg) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java:87]: just formatting

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java (hunk 4 above)
Comment (kishoreg) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java:126]: yes. ZK does allow that. However, the UI will only allow updating the leaf node.

**Comment by mayankshriv** [comment]
@sajjad-moradi You may want to take into account these apis when adding ACL checks.

**Review by mayankshriv — APPROVED** [review]

**Pull request merged**